### PR TITLE
[5448] Display placement details when available

### PR DESCRIPTION
**Pull request by tavy88**
### Context

Display placement details when available and made them exportable in the CSV.

The placement data comes in like this from `hesa_students` table:

```ruby
  placements: [{"school_urn"=>"103560", "placement_days"=>"84"}]
```

However, in order to display the placement details as in the picture, we need to fetch that info from `schools` table.

Analysing the data, it seems like the max amount of placements for a trainee in the latest hesa collection is 5.

#### Magic URNs

Sometimes hesa students will contain placements with urns that are not real. We call these magic urns. E.g

```ruby
[{"school_urn"=>"900000", "placement_days"=>"40"}]
```

In this case we won't have any school records and instead display a hardcoded label.

Ticket: https://trello.com/c/Du0zqTiH

### Changes proposed in this pull request

- Display placement details when available on the trainee page.
- Make the placement data exportable: placement_1 column has the first school URN, placement_2 column has the second school URN and other_placement has the rest of the placements comma separated.

### Guidance to review

Screenshot 2023-05-09 at 13 16 13

Screenshot 2023-05-10 at 17 18 21

Screenshot 2023-05-11 at 13 26 01

Screenshot 2023-05-11 at 13 26 46

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by jhenderson** [comment]
[[M] Show placement data on records and include in export](https://trello.com/c/Du0zqTiH/5448-m-show-placement-data-on-records-and-include-in-export)

File: app/components/placement_details/view.html.erb [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-      custom_value: true,
-      }],
-      editable: editable,
-      has_errors: has_errors,
-      ) do |component| %>
+<%= render MappableSummary::View.new(
+  trainee: trainee, 
+  title: summary_title,
+  rows: rows,
+  editable: editable,
```
Comment (jhenderson) [inline app/components/placement_details/view.html.erb:5]:
```suggestion
  editable: false,
```

**Review by jhenderson — CHANGES_REQUESTED** [review]
Looking really good. Left some minor suggestions.

File: app/models/school.rb [hunk 2]
```diff
@@ -26,6 +26,13 @@
 class School < ApplicationRecord
   include PgSearch::Model
 
+  # We call these magic because they are not real URNs, but they do exist in the
+  # HESA data and we need to be able to display them somehow.
+  MAGIC_URNS = %w[
```
[the remark this answers is not in the document]
Comment (tavy88) [reply app/models/school.rb:31]: Good idea!

**Review by jhenderson — APPROVED** [review]

**Pull request merged**